### PR TITLE
Merge Python 2 and 3 versions of time.pyi

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

File: stdlib/2and3/time.pyi [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
 
 TimeTuple = Tuple[int, int, int, int, int, int, int, int, int]
 
-# ----- variables and constants -----
-accept2dyear = False
-altzone = 0
-daylight = 0
-timezone = 0
-tzname = ...  # type: Tuple[str, str]
+accept2dyear: bool = ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/time.pyi]: This doesn't exist in Python 3 (at least not in 3.6 and master, haven't checked all versions).

File: stdlib/2and3/time.pyi [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
+altzone: int = ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/time.pyi]: You don't need the `= ...` here.

File: stdlib/2and3/time.pyi [hunk 3]
```diff
@@ -67,18 +64,15 @@ else:
...
+def gmtime(secs: Optional[float] = ...) -> struct_time: ...
+def localtime(secs: Optional[float] = ...) -> struct_time: ...
 def mktime(t: Union[TimeTuple, struct_time]) -> float: ...
-def sleep(secs: Union[int, float]) -> None: ...
-def strftime(format: str,
-             t: Union[TimeTuple, struct_time, None] = ...) -> str: ...  # return current time
-def strptime(string: str,
-             format: str = ...) -> struct_time: ...
+def sleep(secs: float) -> None: ...
+def strftime(format: str, t: Optional[Union[TimeTuple, struct_time]] = ...) -> str: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/time.pyi]: The `t` argument is not actually Optional (`time.strftime('%y', None)` throws an error).

Comment (srittau) [reply stdlib/2and3/time.pyi]: Same for asctime() actually. I am changing it to @overloads.

Comment (JelleZijlstra) [reply stdlib/2and3/time.pyi]: Why do you need to? I think `def strftime(format: str, t: Union[TimeTuple, struct_time] = ...)` is fine.

File: stdlib/2and3/time.pyi [hunk 4]
```diff
@@ -1,29 +1,26 @@
...
+# See https://docs.python.org/3/library/time.html
 
 import sys
-from typing import Any, NamedTuple, Tuple, Union
-from types import SimpleNamespace
+from typing import Any, NamedTuple, Tuple, Union, Optional
+if sys.version_info >= (3, 3):
+    from types import SimpleNamespace
 
 TimeTuple = Tuple[int, int, int, int, int, int, int, int, int]
```
Comment (JelleZijlstra) [inline stdlib/2and3/time.pyi]: This should be private.

File: stdlib/2and3/time.pyi (hunk 3 above)
Comment (srittau) [reply stdlib/2and3/time.pyi]: Because `strftime("%y")` and `asctime()` are legal calls (using the current time). Or am I overlooking something?

Comment (JelleZijlstra) [reply stdlib/2and3/time.pyi]: That would be allowed by the code I gave above.

Comment (srittau) [reply stdlib/2and3/time.pyi]: Ah yes, of course, the default value.

File: stdlib/2and3/time.pyi (hunk 4 above)
Comment (srittau) [reply stdlib/2and3/time.pyi]: Agreed, the old Python 2 stub had it private, the Python 3 stub has it public. I wasn't sure whether there wasn't a reason for it to be public. But I can't really think of one.

File: stdlib/2and3/time.pyi (hunk 1 above)
Comment (srittau) [reply stdlib/2and3/time.pyi]: The Python 3.2 documentation marks it as deprecated, Python 3.3 documentation doesn't mention it, so I'll just assume it was removed in 3.3.

**Pull request merged**